src/info/field_spec_info.c

```diff
@@ -89,6 +116,53 @@ FieldSpecInfo FieldSpecInfo_Deserialize(const MRReply *reply) {
     MRReply *error = MRReply_MapElement(reply, IndexError_ObjectName);
     RedisModule_Assert(error);
     info.error = IndexError_Deserialize(error);
+    // attribute used to determine field type
+    info.stats = FieldStats_Deserialize(info.attribute, reply);
 
     return info;
 }
+
+FieldSpecStats FieldStats_Deserialize(const char* type,const MRReply* reply){
+    FieldSpecStats stats = {0};
+    FieldType fieldType = getFieldType(type);
+    switch (fieldType) {
+        case INDEXFLD_T_VECTOR:
+            for(int i = 0; VectorIndexStats_Metrics[i] != NULL; i++){
+                size_t metricValue = MRReply_Integer(MRReply_MapElement(reply, VectorIndexStats_Metrics[i]));
+                VectorIndexStats_GetSetter(VectorIndexStats_Metrics[i])(&stats.vecStats, metricValue);
+            }
+            stats.type = INDEXFLD_T_VECTOR;
+        default:
+            break;
+    }
+    return stats;
+}
+
+FieldType getFieldType(const char* type){
```

Comment:
Can be static, and defined above in the file so we don't need the forward declaration.